src/olympia/addons/widgets.py

```diff
@@ -1,31 +1,32 @@
-from django import forms
 from django.conf import settings
+
+from django import forms
 from django.utils.encoding import force_text
-from django.utils.html import conditional_escape
+from django.utils.html import conditional_escape, format_html
 from django.utils.safestring import mark_safe
 from django.utils.translation import ugettext
 
 from olympia.addons.models import Category
 
 
-class IconWidgetRenderer(forms.RadioSelect.renderer):
-    """Return radiobox as a list of images."""
+class IconTypeSelect(forms.RadioSelect):
 
-    def render(self):
-        """This will output radios as li>img+input."""
+    def render(self, name, value, attrs=None, renderer=None):
         output = []
-        for w in self:
-            value = w.choice_value
-            if value.split('/')[0] == 'icon' or value == '':
-                o = (("<li><a href='#' class='%s'>"
-                      "<img src='%simg/addon-icons/%s-32.png' alt=''>"
-                      "</a>%s</li>") %
-                     ('active' if self.value == w.choice_value else '',
-                      settings.STATIC_URL, w.choice_label, w))
-            else:
-                o = "<li class='hide'>%s</li>" % w
-            output.append(o)
-        return mark_safe(u'\n'.join(output))
+
+        for option in self.options(name, value, attrs):
```

Comment:
Using `format_html_join()` would be neater... and remove the need to `mark_safe()`, I think. But that's an improvement, not something to fix now.

Reply:
I keep forgetting about `format_html_join`, I'll use that. Thanks!